apps/Unciv/install

```diff
@@ -35,16 +35,22 @@ sudo mv -f /tmp/adoptium-archive-keyring.gpg /usr/share/keyrings
 echo " - Creating /etc/apt/sources.list.d/adoptium.list"
 echo "deb [signed-by=/usr/share/keyrings/adoptium-archive-keyring.gpg] https://adoptium.jfrog.io/artifactory/deb $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/adoptium.list >/dev/null
 
-echo " - Installing temurin-8-jdk"
+status "Installing temurin-8-jdk"
 #try to install temurin-8-jdk; if it fails, remove repository to avoid breaking user's system
 (install_packages temurin-8-jdk)
-if [ $? != 0 ];then
+if [ $? != 0 ]; then
   sudo rm -f /etc/apt/sources.list.d/adoptium.list /usr/share/keyrings/adoptium-archive-keyring.gpg
   error "Failed to install temurin packages. Adoptium repository has been removed."
 fi
 
-### Download Unciv.jar
-wget -O "$HOME/Unciv/Unciv.jar" https://github.com/yairm210/Unciv/releases/download/3.18.10-patch1/Unciv.jar || exit 1
+status "Downloading Unciv.jar"
+echo '- writing into /tmp/Unciv.jar'
+wget -O /tmp/Unciv.jar https://github.com/yairm210/Unciv/releases/download/3.18.11-patch1/Unciv.jar \
+    || error 'Failed to download Unciv.jar!'
+echo '- moving /tmp/Unciv.jar to ~/Unciv/Unciv.jar'
+mv /tmp/Unciv.jar ~/Unciv/Unciv.jar || exit 1
+echo '- removing /tmp/Unciv.jar'
+rm -f /tmp/Unciv.jar
```

Comment:
Why are we downloading to /tmp/Unciv.jar, then moving that file to ~/Unciv/Unciv.jar? Is there some safety or performance benefit to doing this?

Reply:
yes. it ensures that the already downloaded `Unciv.jar` files doesn't get corrupt when the update fails. So, even if the download when updating fails, the user can play the previous version without any problem.

Reply:
it can even enable users update the app while playing probably.